src/pad_action_entry.rs

```diff
@@ -48,6 +54,16 @@ impl ToGlib for PadActionEntry {
     type GlibType = gtk_sys::GtkPadActionEntry;
 
     fn to_glib(&self) -> gtk_sys::GtkPadActionEntry {
-        self.0
+        gtk_sys::GtkPadActionEntry {
+            type_: self.type_.to_glib(),
+            index: self.index,
+            mode: self.mode,
+            label: CStr::from_bytes_with_nul(self.label.as_bytes())
+                .expect("Unexpected nul")
+                .as_ptr() as *mut _,
+            action_name: CStr::from_bytes_with_nul(self.action_name.as_bytes())
+                .expect("Unexpected nul")
+                .as_ptr() as *mut _,
```

Comment:
This is still unsafe: The returned `GtkPadActionEntry` borrows the pointers.

I'd simply remove this `ToGlib` impl and implement the one conversion that we actually need manually so that it's actually safe.